src/main/scala/rocket/DCache.scala

```diff
@@ -551,6 +552,20 @@ class DCacheModule(outer: DCache) extends HellaCacheModule(outer) {
     Mux(s2_req.cmd === M_PWR, putpartial,
     Mux(!s2_read, put, atomics))))
 
+  // Drive APROT Bits
+  tl_out_a.bits.user.lift(AMBAProt).foreach { x =>
+    val user_bit_cacheable = edge.manager.supportsAcquireTFast(access_address, a_size)
+
+    x.privileged  := s2_req.dprv === PRV.M || user_bit_cacheable
```

Comment:
so all cacheable accesses are also by definition privileged?

Reply:
The relevant point from the AMBA spec is that if a master cannot generate accurate protection information, it is recommended that we mark the transaction as privileged.  Because caches like the L2 can merge requests from multiple different privilege modes, we can't accurately provide the protection information.